### DISCUSSION TITLE
MIDI Host

### DISCUSSION THREAD

**Discussion by huttokd** [lathoub/Arduino-USBMIDI#24] [Q&A]
Sorry if this is obvious but does this code work for using the Arduino as a host? I want to connect the Arduino to a MIDI keyboard to control various things based on certain keys being pressed. I have an Arduino Leonardo but have seemingly been unable to receive anything from the keyboard using the example codes.

**Comment by lathoub** [comment]
This lib relies heavily on [MIDIUSB](https://github.com/arduino-libraries/MIDIUSB) and is based on [PluggableUSB](https://github.com/arduino/ArduinoCore-API/blob/master/api/PluggableUSB.h). Check these libs for your basic functional requirements.

Leonardo should work fine to receive MIDI messages from your MIDI keyboard.
See this [example](https://github.com/lathoub/Arduino-USBMIDI/blob/master/examples/NoteOnOffEverySec/NoteOnOffEverySec.ino)
